mne/beamformer/_compute_beamformer.py

```diff
@@ -252,51 +207,145 @@ def _prepare_beamformer_input(info, forward, label, picks, pick_ori,
         raise ValueError('Got invalid value for "fwd_norm". Valid '
                          'values are: "dipole", "vertex" or None.')
 
-    return is_free_ori, ch_names, proj, vertno, G
+    return is_free_ori, ch_names, proj, vertno, G, nn
 
 
-def _compute_beamformer(method, G, Cm, reg, n_orient, weight_norm,
-                        pick_ori, reduce_rank, rank, is_free_ori,
-                        inversion=None):
-    """Compute a spatial filter (LCMV or DICS)."""
-    # Tikhonov regularization using reg parameter d to control for
-    # trade-off between spatial resolution and noise sensitivity
-    if method == 'lcmv':
-        Cm_inv, d = _reg_pinv(Cm.copy(), reg)
-    elif method == 'dics':
-        Cm_inv, _ = _reg_pinv(Cm, reg, rcond='auto')
+def _shortcut(Wk, Gk, Cm_inv_sq, reduce_rank, nn):
```

Comment:
Maybe use a more descriptive name, something like ``_norm_weights`` or such? 
Is it planned to use the following code more often? So far this function is only used once and it might be easier for future looks at the code to not have to jump back and forth?